### DISCUSSION TITLE
How to get very long KV cached

### DISCUSSION THREAD

**Discussion by pj-ml** [sgl-project/sglang#102] [Q&A] [answered]
Before I start, I just want to say that I stumbled across this repo today, and I love it!

I have a long common prefix (like 5000 tokens) and then add questions about the long prefix at the end. I have about 15 or so questions. How can I get the 5000 token prefix KVs cached?

Even a hacky way as a temporary solution would be much appreciated!

**Comment by merrymercy** [comment, marked as answer]
@pj-ml Thanks for your interest. SGLang runtime is just designed for this and can greatly accelerate your workloads.
Please see https://github.com/sgl-project/sglang/issues/106#issuecomment-1915875006
